packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -666,9 +754,13 @@ export class SeedlessOnboardingController<EncryptionKey> extends BaseController<
       return isExpiredPwd;
     };
 
-    return options?.skipLock
-      ? await doCheck()
-      : await this.#withControllerLock(doCheck);
+    return await this.#executeWithTokenRefresh(
+      async () =>
+        options?.skipLock
+          ? await doCheck()
```

Comment:
```suggestion
          ? await doCheckIsPasswordExpired()
```

Suggesting changing name of `doCheck` to `doCheckIsPasswordExpired`

Reply:
also inside the doCheck function we are caching the result but not using the cached result on calling the function again, is there any reason for that?

Reply:
@himanshuchawla009 you're right, the cache check should be inside, will update this